File: 프로그래머스/lv0/120812. 최빈값 구하기/최빈값 구하기.java
class Solution {
    public int solution(int[] array) {
        int answer = 0;
        
        int[] cnt = new int[1001];
        
        for(int i=0;i<array.length;i++){
            cnt[array[i]]++;
        }
        
        int max = 0;
        int max_idx = 0;
        
        for(int j=0;j<cnt.length;j++){
            if(cnt[j]>max){
                max = cnt[j];
                max_idx = j;
            }    
        }
        
        int count = 0;
        for(int i=0;i<cnt.length;i++){
            if(cnt[i] == max) count++;
        }
        
        if(count > 1) return -1;
        answer = max_idx;
        return answer;
    }
}

File: 프로그래머스/0/120812. 최빈값 구하기/최빈값 구하기.js
function solution(array) {
    let m = new Map();
    for (let n of array) m.set(n, (m.get(n) ?? 0)+1);
    m = [...m].sort((a,b)=>b[1]-a[1]);
    return m.length === 1 || m[0][1] > m[1][1] ? m[0][0] : -1;
}